der/src/tag.rs

```diff
@@ -476,6 +568,29 @@ mod tests {
         }
     }
 
+    #[test]
+    fn decoding() {
+        //
+        assert_eq!(
+            ErrorKind::TagNumberInvalid,
+            Tag::from_der(&hex!("FF03"))
```

Comment:
Suggested test values:
1. 5f818001 = Tag::Application(0x4001)
2. 9f81808001 =  Tag::ContextSpecific(0x200001)
3. Whatever the heck tag 2^32-1 is 😃
4. df8f8f8f8f0f = Invalid (overflow)
5. 5f9e = Invalid (too small for a multi-byte tag)
6. More?

Apologies if some of the above are wrong, I was writing them up by hand and I'm not an ASN.1 expert.